scribejava-core/src/main/java/com/github/scribejava/core/model/OAuthRequest.java

```diff
@@ -128,32 +128,61 @@ public void addParameter(String key, String value) {
     }
 
     /**
-     * Set boundary of multipart request
+     * Set subtype and boundary of multipart request
      *
-     * @param boundary can be any string
+     * @param subtype subtype of multipart content-type
+     * @param boundary can be any string between 1 and 70 characters long using alphanumeric characters and any of
+     *  "'" / "(" / ")" / "+" / "_" / "," / "-" / "." / "/" / ":" / "=" / "?" / " ", but must not end with white space
      */
-    public void initMultipartBoundary(String boundary) {
-        multipartPayload = new MultipartPayload(boundary == null
-                ? Long.toString(System.currentTimeMillis())
+    public void initMultipartPayload(String subtype, String boundary) {
+        multipartPayload = new MultipartPayload(subtype == null ? "mixed" : subtype, boundary == null
+                ? "---------------------------" + Long.toString(System.currentTimeMillis())
```

Comment:
what for is this string "---------------------------" here?

Reply:
I don't think it's technically necessary, but both Firefox and Chrome do it in some form or another for their multipart form-data requests (which is what I needed and tried to emulate and got working with this fix), so I thought it may be good practice. But I don't know why they do it exactly. I think it might very well be possible there's more to it than just a fixed number of dashes, too. So it can probably be removed without losing anything.